### ISSUE TITLE
avrdude documentation improvement 

### ISSUE THREAD

**Issue by mcuee** [avrdudes/avrdude#1674]
https://avrdudes.github.io/avrdude/7.3/avrdude.html

Some of the information are a bit outdated.
1) Windows installation -- maybe we want to point to the Wiki. We do not really recommend using Cygwin. We do not really support MinGW.org now (website down) but rather MSYS2 mingw project.
https://avrdudes.github.io/avrdude/7.3/avrdude_34.html#Windows-Installation

2) Windows Configuration File Location -- I guess we can forget about 16bit Windows NT now. avrdude only works under Windows 7 and later.
https://avrdudes.github.io/avrdude/7.3/avrdude_37.html#Windows-Configuration-File-Location

3) Windows port name -- no more parallel port support.
https://avrdudes.github.io/avrdude/7.3/avrdude_39.html#Serial-Ports --> not so sure we want to mention libserialport improvement (only for MinGW build).
https://avrdudes.github.io/avrdude/7.3/avrdude_40.html#Parallel-Ports

4) We need to remove Windows Parallel Port info here as well.
https://avrdudes.github.io/avrdude/7.3/avrdude_4.html#Option-Descriptions

**Comment by mcuee** [comment]
I think some of the info below about serialupdi needs updates as well.
https://avrdudes.github.io/avrdude/7.3/avrdude_21.html#SerialUPDI-Programmer

@dbuchwald 
Would you be able to help here? Thanks.

**Comment by dbuchwald** [comment]
Sorry, took me a while, but you can find updates to SerialUPDI documentation in PR #1689

**Comment by mcuee** [comment]
> Sorry, took me a while, but you can find updates to SerialUPDI documentation in PR #1689

You are very fast already. Thanks.
PR #1689 looks good to me.

**Comment by mcuee** [comment]
> I think some of the info below about serialupdi needs updates as well. https://avrdudes.github.io/avrdude/7.3/avrdude_21.html#SerialUPDI-Programmer
> 
> @dbuchwald Would you be able to help here? Thanks.

Done. PR #1689 has been merged.

**Comment by mcuee** [comment]
@stefanrueger @ndim @dl8dtl @MCUdude 

My suggestion is totally remove Section A from the documentation as it does not provide much value to the user. Windows section is completely outdated as well.

https://avrdudes.github.io/avrdude/7.3/avrdude_23.html#Platform-Dependent-Information

**Comment by stefanrueger** [comment]
> totally remove Section A 

Hugely controversial b/c I just spend a couple of hours the other day to *add* info in USB permissions :smile:. Check out the current [live documentation](https://github.com/avrdudes/avrdude/blob/main/avrdude.pdf). But yes, I agree to remove outdated or wrong information. Only problem: I don't know what is. But you do @mcuee and could help?

**Comment by stefanrueger** [comment]
> Windows section is completely outdated as well

I disagree. The location of configuration files, eg, should be correct. And how should a user know where their avrdude.rc goes?

**Comment by mcuee** [comment]
> > Windows section is completely outdated as well
> 
> I disagree. The location of configuration files, eg, should be correct. And how should a user know where their avrdude.rc goes?

Good point. We can keep this one and remove others.

Sorry but no, I can not help to create the PR myself. I do not know anything and refuse to learn anything about Tex.
=

**Comment by ndim** [comment]
The two things texinfo and TeX have in common is they both are devades old and they both use the letters tex.

**Comment by mcuee** [comment]
> The two things texinfo and TeX have in common is they both are decades old and they both use the letters tex. But point taken, texinfo is not a nice thing to handle.

Thanks for the correction. Somehow I mixed the two. Tex was huge last time I played Linux on a 486 machine so the first thing is to remove it. Somehow we need to install Tex in order to build avrdude documentation.

You can see we need the following three packages installed under Linux to build avrdude documentation.

          texinfo
          texlive
          texi2html

**Comment by mcuee** [comment]
For the Unix section, at least the section on Installation needs to be updated now that we support CMake. But I tend to think it is a good idea to remove the installation section.

**Comment by mcuee** [comment]
> > Windows section is completely outdated as well
> 
> I disagree. The location of configuration files, eg, should be correct. And how should a user know where their avrdude.rc goes?

@stefanrueger 
It is already in the main section. So better remove as well.
https://avrdudes.github.io/avrdude/7.3/avrdude_10.html#Configuration-Files

**Comment by stefanrueger** [comment]
@mcuee Happy to remove the installation section. I still would like a link with info how to install. What's my best best?
BTW, the project needs the info what's outdated and what to replace it with. Seeing that you know way more about the platform-dependent issues that everyone else on the team, it would be great if you could contribute that info. Once the info is there it's relatively easy to drop that into the documentation.

**Comment by mcuee** [comment]
> @mcuee Happy to remove the installation section. I still would like a link with info how to install. What's my best best? BTW, the project needs the info what's outdated and what to replace it with. Seeing that you know way more about the platform-dependent issues that everyone else on the team, it would be great if you could contribute that info. Once the info is there it's relatively easy to drop that into the documentation.

Let me work on the changes to the Appendix over the weekend.

**Comment by mcuee** [comment]
Ideas of changes to Appendix A using the latest pdf as the reference.
https://github.com/avrdudes/avrdude/blob/main/avrdude.pdf 

**A.1.1 --> A.1.1 Unix Installation** 
To remove

**A.1.2 --> A.1.2 Unix Configuration Files**
Will need to keep as it is referenced from Ch 4 "4 Configuration Files".

- To add macOS Homebrew and macPorts.
- To add OpenBSD and NetBSD

** A.1.3 Unix Port Names **

- To mention Parallel port is not supported by macOS
- To mention that the user may need to check the USB serial port name
- To mention that libserialport enabled installation has alternatives to mention the port name.

** A.1.4 Unix USB Permissions **

- To add a reference for macOS, USB HID device needs to use HIDAPI and not libusb. When using libusb, need to have no kernel driver attached to the USB interface of interests.
- To add information for OpenBSD and NetBSD if I can find a good reference.

** A.1.5 Unix Documentation **

No changes needed.

**Comment by mcuee** [comment]
Ideas of changes to Appendix A using the latest pdf as the reference.
https://github.com/avrdudes/avrdude/blob/main/avrdude.pdf

**A.2.1 Installation**

- To remove.

**A.2.2 Windows Configuration Files**

- To keep.

**A.2.3 Windows Port Names**

- To include the alternative way to name a serial port if libserialport is used.
- To mention that Parallel port is not supported
- For USB HID device, HIDAPI needs to be used. PICkit 2 is an exception which uses native Windows API.
- For USB device, libusb-win32 or libusb-compat-0.1 or libusb or avrdude-libusb need to be used. Mention the prefer driver is WinUSB even though libusb0.sys or libusbK.sys should work as well.

**Comment by mcuee** [comment]
Further improvements for Appendix A.

Need to see if we can incorporate some of the info from here. Maybe post avrdude-8.0 release because of time constraints.
https://github.com/avrdudes/avrdude/wiki/Known-limitations-of-avrdude

**Comment by stefanrueger** [comment]
Thanks @mcuee of breaking this down in smaller tasks

> To add macOS Homebrew and macPorts.
> To add OpenBSD and NetBSD

If this is going to happen the project needs the text of what to add!

**Comment by stefanrueger** [comment]
Same with all the other points. I won't be able to do that.

**Comment by stefanrueger** [comment]
Have rerun the boxed examples in the documentation and put the new look and feel to work.

@mcuee I have replaced the Installation sections with a reference to the wiki. For the rest I need *text*.

**Comment by mcuee** [comment]
I will see what I can do over the weekend. Still recovering from Covid-19.

**Comment by mcuee** [comment]
@stefanrueger 

I looked at the current avrdude.pdf, I think it is quite an improvement over the 7.3 release version. Thanks.

I will suggest a few changes below.

Two must-have changes for Windows.
1) Remove Section `A.2.3.2 Windows Parallel Ports` will do. Or we can keep the section and say that `Parallel Ports are not suppoted by avrdude under Windows.` 

2) Remove item 5 of Section A.2.2.2 (no more 16bit Windows subsytem.

```
A.2.2.2 Windows Configuration File Location
AVRDUDE on Windows has a different way of searching for the system and user configuration
files. Below is the search method for locating the configuration files:
1. Only for the system configuration file: <directory from which application
loaded>/../etc/avrdude.conf
2. The directory from which the application loaded.
3. The current directory.
4. The Windows system directory. On Windows NT, the name of this directory is SYSTEM32.
<del> 5. Windows NT: The 16-bit Windows system directory. The name of this directory is SYSTEM. </del>
6. The Windows directory.
7. The directories that are listed in the PATH environment variable.
```

**Comment by mcuee** [comment]
@stefanrueger 

Optional changes.
1) Add a note in `Section A.1.3 Unix Port Names` saying that `Please take note that parallel port is not supported by macOS`.

**Comment by mcuee** [comment]
@stefanrueger 

For Appendix B, I'd like to suggest the changes for the first sentence.
```
In general, please report any bugs encountered via
https://github.com/avrdudes/avrdude/issues.
```
-->
```
In general, please report any bugs encountered via
https://github.com/avrdudes/avrdude/issues.

Please also consult the Wiki to learn more about avrdude.
https://github.com/avrdudes/avrdude/wiki

In paticular the FAQ and  the known limitation of avrdude are worth reading.
[Known limitations of avrdude](https://github.com/avrdudes/avrdude/wiki/Known-limitations-of-avrdude)
[FAQ](https://github.com/avrdudes/avrdude/wiki/FAQ)

```

**Comment by mcuee** [comment]
> @stefanrueger
> 
> I looked at the current avrdude.pdf, I think it is quite an improvement over the 7.3 release version. Thanks.
> 
> I will suggest a few changes below.
> 
> Two must-have changes for Windows.
> 
> 1. Remove Section `A.2.3.2 Windows Parallel Ports` will do. Or we can keep the section and say that `Parallel Ports are not suppoted by avrdude under Windows.`
> 2. Remove item 5 of Section A.2.2.2 (no more 16bit Windows subsytem.
> 
> ```
> A.2.2.2 Windows Configuration File Location
> AVRDUDE on Windows has a different way of searching for the system and user configuration
> files. Below is the search method for locating the configuration files:
> 1. Only for the system configuration file: <directory from which application
> loaded>/../etc/avrdude.conf
> 2. The directory from which the application loaded.
> 3. The current directory.
> 4. The Windows system directory. On Windows NT, the name of this directory is SYSTEM32.
> <del> 5. Windows NT: The 16-bit Windows system directory. The name of this directory is SYSTEM. </del>
> 6. The Windows directory.
> 7. The directories that are listed in the PATH environment variable.
> ```

@stefanrueger 

I take the liberty to commit two changes to git main for the above changes.

**Comment by stefanrueger** [comment]
Thanks, very helpful.

**Comment by mcuee** [comment]
@stefanrueger and @MCUdude 

Two minor updates for macOS in PR#1902. Please review. Thanks.
* #1902

**Comment by mcuee** [comment]
I think we have done the improvements.

**Comment by stefanrueger** [comment]
> I think we have done the improvements.

I don't think so, eg, I don't see [this](https://github.com/avrdudes/avrdude/issues/1674#issuecomment-2277372732) and [that](https://github.com/avrdudes/avrdude/issues/1674#issuecomment-2277384328) fully addressed. 

It sounds like it should be, but I for one don't know know how and no one else has been chipping in. For example 
  - "the user may need to check the USB serial port name" is insufficient info for me to put in a manual. I really don't know what that means. Is that a graphic user interface with a box they check? Do they need to investigate something?
  -  "libserialport enabled installation has alternatives to mention the port name". I can write something to the effect of If AVRDUDE was compiled with the `@code`{libserialport} library ports will also have alternative names that are derived form the used USB to serial chip, e.g., XYZ but I am missing a crucial example here that was taken from a real Windows system.
  - " USB HID device needs to use HIDAPI and not libusb": Is that advice to the user to recompile? How would they know which device is USB HID? Which errors would they see? 
 - "When using libusb, need to have no kernel driver attached to the USB interface of interests." Remember this is a user manual. How would they know which library avrdude uses? How would they see a problem? Which commands would they issue to figure out kernel drivers being attached? Etc

I am not trying to be difficult. I simply don't know and think someone who does should give it a go. 

Reflecting on above, some could go into the Troubleshooting session, and some into the wiki for how to compile.

**Comment by mcuee** [comment]
Review of the current documentation by Claude Sonnet 4.6 Slow (free version of Claude).
+++++++++++++++++

Here's a clean summary ready to paste into the issue:

**Documentation review against 8.1 release and current main**
Reviewed the 8.1 HTML docs and current avrdude.texi against issue #1674. Several items from #1674 were fixed; the following remain open.

**Still open from #1674**

**1. A.2.3.1 Windows Serial Ports — still one sentence unchanged since 7.3:**

"When you select a serial port (i.e. when using an STK500) use the Windows serial port device names such as: com1, com2, etc."

Suggested additions:

- How to find the assigned COM port number in Windows Device Manager (Ports → COM & LPT) when a USB-serial adapter is used
- Cross-reference to the -P ?s command already documented in the -P option description
- Cross-reference to the libserialport named-adapter support (-P ch340, -P ft232r, etc.) already documented in the -P option description

**2. libusb vs HIDAPI / kernel driver guidance — still no troubleshooting entries for:**

- Windows: programmer not found / access denied → install WinUSB driver via Zadig
- Windows: device works in Atmel Studio but not AVRDUDE → HID-mode devices need HIDAPI support compiled in
- Linux: detach kernel driver before using libusb-based programmers

These were flagged in #1674 as needing concrete examples before someone can write them. The wiki FAQ may be the better first home.

**3. Typos (trivial fixes)**

- Introduction, CH341A paragraph: "They former two feature" → "The former two feature"
- Introduction, linuxspi paragraph: "occuring" → "occurring"
- 2.1 Option Descriptions, -e option: "commad" → "command"
- A.2.3.2 Windows Parallel Ports: "using programmer on a parallel port" → "using a programmer on a parallel port"

**4. Platform consistency**

- Introduction, PPI paragraph: states PPI/parallel port support without qualification, contradicting the new A.2.3.2 note that Windows does not support it. Suggest adding "on Linux and BSD systems" or equivalent.
- 2.1 Option Descriptions, -P option: "This can be a parallel, serial, spi or linuxgpio connection" — parallel should be qualified as Linux/BSD only, consistent with A.2.3.2.

**5. Stale troubleshooting entries**

- Cygwin serial workaround entry (avrdude: serial_open(): can't set attributes) — orphaned now that Installation no longer mentions Cygwin. Suggest removing.
- ATxmega A1/A2/A3 PDI entry dated "2010 Q1" — AVR Dragon is discontinued, these parts are out of production. Suggest removing.
- libusb ZLP bug entry — affects "Linux with kernel < 2.6.31 and libusb >= 1.0.0 < 1.0.3". Linux 2.6.31 is from 2009; no supported distro ships it. Suggest removing.

**6. Minor**

- "1.1 History and Credits": Savannah repository not noted as archived. Suggest adding that Savannah is now read-only and GitHub is the sole active location.
- "Introduction", linuxgpio paragraph: links to a 2013 third-party blog post ("kolev.info") as the primary tutorial. Fragile external dependency; consider moving content to the wiki and linking there instead.

**Comment by stefanrueger** [comment]
Ah, good. Thanks @mcuee.

Yes, we can do a few low-hanging fruits (eg, typos). As to the rest: this needs text by someone familiar with Windows, libusb, HIDAPI etc. @mcuee?

>  AVR Dragon is discontinued, these parts are out of production. Suggest removing

No! Avrdude always has tried to be backwards compatible. And it really should. The fact that a programmer is discontinued doesn't mean it will evaporate from the drawers of all the 1000's of users who once bought it. My view is that they still have the programmer and they still might want to use it!

**Comment by mcuee** [comment]
@stefanrueger 

I will try to prepare a simple write-up this week (15-21 June 2026 week).

**Comment by mcuee** [comment]
@stefanrueger 

I have updated Wiki FAQ entry here to address some of the issues with the documentations.
https://github.com/avrdudes/avrdude/wiki/FAQ

I have also updated libusb Windows page here.
https://github.com/avrdudes/avrdude/wiki/The-story-on-libusb-for-Windows

**Comment by stefanrueger** [comment]
@mcuee Great! That's useful

Has [above review](https://github.com/avrdudes/avrdude/issues/1674#issuecomment-4573262995)'s Section 2 ( libusb vs HIDAPI / kernel driver guidance) been addressed in `avrdude.text`?

**Comment by mcuee** [comment]
> [@mcuee](https://github.com/mcuee) Great! That's useful
> 
> Has [above review](https://github.com/avrdudes/avrdude/issues/1674#issuecomment-4573262995)'s Section 2 ( libusb vs HIDAPI / kernel driver guidance) been addressed in `avrdude.text`?

Not yet. I will add more info in the WiFi FAQ first to addrees the issue.

**Comment by mcuee** [comment]
@stefanrueger 

I have added the following note to the FAQ. The whole Wiki FAQ entry should be able to address the libusb/hidapi selection issue.
https://github.com/avrdudes/avrdude/wiki/FAQ#shall-i-use-hidapi-or-libusb-for-usb-hid-device
```
Example of USB HID based CMSIS-DAP compliant programmers. There is a known issue #1221 with these programmers when using libusb. You should use hidapi.
1) Atmel ICE and Power Debugger -- high speed USB HID device with HID report and packet size of 512 bytes
2) PIckit 4 in AVR mode-- high speed USB HID device with HID report and packet size of 64 bytes
3) SNAP in AVR mode -- full speed USB HID device with HID reprot and packet size of 64 bytes
```
